components/docker-up/BUILD.yaml

```diff
@@ -40,6 +40,10 @@ packages:
       - :bin-runc-facade
     config:
       commands:
+        - ["mkdir", "unarchived"]
+        - ["tar", "-zxvf", "components-docker-up--bin-docker-up/docker.tgz", "-C", "unarchived"]
+        - ["cp", "-vaR", "unarchived/docker/.", "."]
+        - ["rm", "-r", "unarchived"]
```

Comment:
@akosyakov wdyt of changing this PR title and description? My understanding is you are trying to get `gp rebuild` working, need the docker CLI, but docker support in rootless images (as a feature) is not required/can be decoupled from this effort.

Suggestions:
1. As @atduarte suggested, [treat embedded docker bin as a portable app](https://www.notion.so/Support-Docker-in-rootless-images-159557fdd44c4c38892c180df9978ee7?d=003373ac66724ff393837fefef3c0816#cc4d8f4b00794f01a035607c96804c32). For example, like you mentioned, instead of persisting to `unarchived`, do so in `.supervisor/docker/bin`, and then have `gp rebuild` expect it to be there. This way:
   * The dependency for `gp rebuild` on the docker CLI does not break or change the end-user experience for docker CLI
   * The IDE team is unblocked
2. Inbox an issue to the Workspace Team, to consider installing docker CLI as part of `docker-up` like you have now, so customers get `docker support in rootless images`. At this time, we could also consider deleting `.supervisor/docker/bin`. _This would be a lower priority given our current priorities._

Regarding deploy for the `docker-up` change, we prboably have to ship a new workspace cluster (I do not see `docker-up` as a version that can be overridden in `ide-config` in the application clusters). 
* `registry-facade` has a specific `docker-up` in its configmap for `staticLayer` :

```
 "staticLayer": [
      {
        "ref": "eu.gcr.io/gitpod-core-dev/build/supervisor:commit-1e24370e8e1ee19ca3b8d9521e1c205269978680",
        "type": "image"
      },
      {
        "ref": "eu.gcr.io/gitpod-core-dev/build/workspacekit:commit-243207eb85e1ac9ada023fc6ba3a4e3118d49d5e",
        "type": "image"
      },
      {
        "ref": "eu.gcr.io/gitpod-core-dev/build/docker-up:commit-243207eb85e1ac9ada023fc6ba3a4e3118d49d5e",
        "type": "image"
      }
    ],
```

You shared an [internal concern](https://www.notion.so/Support-Docker-in-rootless-images-159557fdd44c4c38892c180df9978ee7?d=003373ac66724ff393837fefef3c0816#cc4d8f4b00794f01a035607c96804c32) about docker, docker-compose, and runc versions.
* We have a fork of docker-compose, where we override the MTU [here](https://github.com/gitpod-io/compose/pull/1)
* docker and runc updates [here](https://github.com/gitpod-io/gitpod/pull/11976) were for currency and many issues related to clone3

Reply:
I have another idea right now to let supervisor to do docker build/run under sudo, then it can activate docker daemon and use shipped cli even if a user don't have sudo access.

I probably can go with the variant one as well, but it adds complexity since we need to ship docker twice, and it does not strike me that shipping random docker cli has valid usage. As you mentioned version of docker-compose already pinned and overrides whatever user has in the workspace image.

I was thinking to suggest removing docker-up from static layer and instead allow ide-service to deliver it. Then we could agree on something simple like merging this PR and if it does not work revert it with and IDE deployment. With that we way we would learn faster and does not need to deliver docker twice.